ugsync/src/main/java/org/apache/ranger/unixusersync/process/PolicyMgrUserGroupBuilder.java

```diff
@@ -801,7 +811,55 @@ private void delXUserGroupInfo(XUserInfo aUserInfo, XGroupInfo aGroupInfo) {
 		}
 
 	}
-	
+
+	private MUserInfo addMUser(String aUserName, List<String> groups) {
+		MUserInfo ret = null;
+		MUserInfo userInfo = new MUserInfo();
+
+		userInfo.setLoginId(aUserName);
+		userInfo.setFirstName(aUserName);
+		userInfo.setLastName(aUserName);
+
+		if (isUserInAdminGroup (aUserName, groups)) {
+			String[] userRoleList = { "ROLE_SYS_ADMIN" };
+			userInfo.setUserRoleList(userRoleList);
+		}
+
+		if (authenticationType != null && AUTH_KERBEROS.equalsIgnoreCase(authenticationType) && SecureClientLogin.isKerberosCredentialExists(principal, keytab)) {
+			try {
+				Subject sub = SecureClientLogin.loginUserFromKeytab(principal, keytab, nameRules);
+				final MUserInfo result = ret;
+				final MUserInfo userInfoFinal = userInfo;
+				ret = Subject.doAs(sub, new PrivilegedAction<MUserInfo>() {
+					@Override
+					public MUserInfo run() {
+						try {
+							return getMUser(userInfoFinal, result);
+						} catch (Exception e) {
+							LOG.error("Failed to add User : ", e);
+						}
+						return null;
+					}
+				});
+				return ret;
+			} catch (Exception e) {
+				LOG.error("Failed to Authenticate Using given Principal and Keytab : " , e);
```

Comment:
This is not an error. This is authentication failure, and you should write to audit log. But it is not a server error.